lib/trino-parquet/src/main/java/io/trino/parquet/writer/ParquetWriters.java

```diff
@@ -202,6 +212,9 @@ private static PrimitiveValueWriter getValueWriter(ValuesWriter valuesWriter, Ty
             verifyParquetType(type, parquetType, OriginalType.TIME_MICROS);
             return new TimeMicrosValueWriter(valuesWriter, parquetType);
         }
+        if (type instanceof TimestampType && parquetType.getPrimitiveTypeName().equals(PrimitiveType.PrimitiveTypeName.INT96)) {
+            return new Int96TimestampValueWriter(valuesWriter, type, parquetType, parquetTimeZone);
+        }
```

Comment:
`timestamp with time zone` does not apply  for Hive, so I am assuming that there shouldn't be created a corresponding  `Int96TimestampTzValueWriter`. I'm just leaving the comment here in case somebody has a different opinion.

Reply:
Yes, we don't try to handle `timestamp with time zone` in similar code for the default parquet writer in `FieldSetterFactory`